main.go

```diff
@@ -152,35 +152,42 @@ func main() {
 	}
 
 	// Write function
+
+	docs_written := 0
 	t := time.NewTicker(time.Second)
 	defer t.Stop()
-	docs_written := 0
-	for numDocs < 0 || docs_written < numDocs {
-		select {
-		// when doneChan is closed, receive immediately returns the zero value
-		case <-doneChan:
-			log.Printf("done")
-			os.Exit(0)
-		case <-t.C:
-			for i := 0; i < wps; i++ {
-				// TODO: move doc generation out of this loop into a go routine that pre-generates them
-				docs, err := generator.GenerateDocs(batchSize, destination, generatorIdentifier, idMode)
-				if err != nil {
-					log.Printf("document generation failed: %v", err)
-					os.Exit(1)
-				}
-				go func(i int) {
-					if err := d.SendDocument(docs); err != nil {
-						log.Printf("failed to send document %d of %d: %v", i, wps, err)
+	if mode == "both" || mode == "add" {
+		for numDocs < 0 || docs_written < numDocs {
+			select {
+			// when doneChan is closed, receive immediately returns the zero value
+			case <-doneChan:
+				log.Printf("done")
+				os.Exit(0)
+			case <-t.C:
+				for i := 0; i < wps; i++ {
+					// TODO: move doc generation out of this loop into a go routine that pre-generates them
+					docs, err := generator.GenerateDocs(batchSize, destination, generatorIdentifier, idMode)
+					if err != nil {
+						log.Printf("document generation failed: %v", err)
+						os.Exit(1)
 					}
-				}(i)
-				docs_written = docs_written + batchSize
+					go func(i int) {
+						if err := d.SendDocument(docs); err != nil {
+							log.Printf("failed to send document %d of %d: %v", i, wps, err)
+						}
+					}(i)
+					docs_written = docs_written + batchSize
+				}
+				// TODO: this does not guarantee that the writes have finished
 			}
-			// TODO: this does not guarantee that the writes have finished
 		}
 	}
 
-	if mode == "patch" {
+	if mode == "both" || mode == "patch" {
+		if mode == "patch" {
+			// must explicitly set number of docs so updates are applied evenly across document keys
+			generator.SetMaxDoc(numDocs)
```

Comment:
We should add NUM_DOCS as a strict requirement if PATCH mode is specified

Reply:
Yes, good call